debezium-embedded/src/main/java/io/debezium/embedded/EmbeddedEngine.java

```diff
@@ -971,6 +974,7 @@ public boolean stop() {
         Thread thread = this.runningThread.getAndSet(null);
         if (thread != null) {
             try {
+                // Making sure the event source coordinator has enough time to shut down before forcefully stopping it
```

Comment:
The defautl wait time is pretty large. In this case there should be a log message informing the user about the wait time.

Reply:
Done.